File: ScriptBeeClient/src/app/services/plugin/marketplace-project.ts
import { PluginKind } from './plugin';

export type MarketplaceProjectType = 'Plugin' | 'Bundle';

export interface MarketplaceProject {
  id: string;
  name: string;
  type: MarketplaceProjectType;
  authors: string[];
  description: string;
  versions: PluginVersion[];
}

export interface PluginVersion {
  version: string;
  extensionPointVersions: ExtensionPointVersion[];
  installed: boolean;
}

export interface ExtensionPointVersion {
  kind: PluginKind;
  version: string;
}

File: ScriptBeeWebApp/EndpointDefinitions/DTO/MarketplaceProject.cs
namespace ScriptBeeWebApp.EndpointDefinitions.DTO;

public sealed record MarketplaceProject(
    string Id,
    string Name,
    string Type,
    string Description,
    List<string> Authors,
    List<PluginVersion> Versions
)
{
    public const string PluginType = "Plugin";
    public const string BundleType = "Bundle";
}

public sealed record PluginVersion(string Version, bool Installed);
